cluster-autoscaler/utils/backoff/exponential_backoff.go

```diff
@@ -87,14 +88,21 @@ func (b *exponentialBackoff) Backoff(nodeGroup cloudprovider.NodeGroup, nodeInfo
 		duration:            duration,
 		backoffUntil:        backoffUntil,
 		lastFailedExecution: currentTime,
+		errorInfo:           errorInfo,
 	}
 	return backoffUntil
 }
 
-// IsBackedOff returns true if execution is backed off for the given node group.
-func (b *exponentialBackoff) IsBackedOff(nodeGroup cloudprovider.NodeGroup, nodeInfo *schedulerframework.NodeInfo, currentTime time.Time) bool {
+// BackoffStatus returns whether the execution is backed off for the given node group and error info when the node group is backed off.
+func (b *exponentialBackoff) BackoffStatus(nodeGroup cloudprovider.NodeGroup, nodeInfo *schedulerframework.NodeInfo, currentTime time.Time) Status {
 	backoffInfo, found := b.backoffInfo[b.nodeGroupKey(nodeGroup)]
-	return found && backoffInfo.backoffUntil.After(currentTime)
+	if !found || !backoffInfo.backoffUntil.After(currentTime) {
```

Comment:
nit: `backoffInfo.backoffUntil.Before(currentTime)` seems more readable.

Reply:
The reason I didn't do it was to make it the same as before, as using Before will be different in the case backoffUntil == currentTIme.
But I have changed it to the suggested code since the chances to happen is almost zero (except for the tests which I have made some changes), and also it is similar to the code in Backoff()